src/pylint/rules/magic_value_comparison.rs

```diff
@@ -0,0 +1,46 @@
+use num_bigint::BigInt;
+use rustpython_ast::{Constant, Expr, ExprKind};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violations;
+
+fn is_magic_value(constant: &Constant) -> Option<String> {
+    match constant {
+        Constant::Int(value) => {
+            if vec![-1, 0, 1]
+                .iter()
+                .map(|&v| BigInt::from(v))
+                .any(|v| v == *value)
+            {
+                None
```

Comment:
I think the following would be more idiomatic:

```rust
    if let Ok(value) = value.try_into() {
        if matches!(value, -1 | 0 | 1) {
            return None;
        }
    }
```

Reply:
Agreed!